### PR TITLE
Update README and key docs

### LINKED ISSUE #104: Update README
#### Summary
Update README to reflect current status of the crate. Give an overview of the signature scheme and examples on how to use this crate.

### PR DESCRIPTION
**Pull request by HDauven**
Resolves #104 #105

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +15,43 @@ be found [here](https://eprint.iacr.org/2019/458.pdf).
 For a reference to the algorithm, please see the 
 [docs](https://app.gitbook.com/@dusk-network/s/specs/proposals/rfc/rfc-17).
```
Comment (moCello) [inline README.md:16]: Does this link still work?

File: src/keys/public.rs [hunk 2]
```diff
@@ -4,14 +4,49 @@
...
-/// Structure repesenting a [`NotePublicKey`]
+/// Structure repesenting a [`NotePublicKey`] an extended point on the JubJub
+/// curve [`JubJubExtended`]. This public key allows for the verification of
+/// signatures created with its corresponding secret key without revealing the
+/// secret key itself.
+///
+/// ## Fields
+///
+/// - `0`: The [`JubJubExtended`] point representing the public key on the
+///   JubJub curve. This field is not directly accessible due to being private.
```
Comment (moCello) [inline src/keys/public.rs:32]: I'm not sure if 'Fields' make so much sense here. I find the `0` very confusing.
In my opinion it's enough to say what you wrote above, that the `NotePublicKey` is a `JubJubExtended`

File: src/keys/public.rs [hunk 3]
```diff
@@ -4,14 +4,49 @@
...
+/// use dusk_bls12_381::BlsScalar;
+/// use rand::rngs::StdRng;
+/// use rand::SeedableRng;
+///
+/// let mut rng = StdRng::seed_from_u64(12345);
+/// let note_secret_key = NoteSecretKey::random(&mut rng);
+///
+/// let pk = NotePublicKey::from(&sk);
+/// let pk_pair: NotePublicKeyPair::from(&sk);
+/// // `signature` is now a secure representation of the signed message.
```
Comment (moCello) [inline src/keys/public.rs:48]: lingering comment?

**Review by moCello — APPROVED** [review]
Overall LGTM, little nits here and there

File: README.md (hunk 1 above)
Comment (HDauven) [reply README.md:16]: Yup!

File: src/keys/public.rs (hunk 3 above)
Comment (HDauven) [reply src/keys/public.rs:48]: Nope! It was meant as a comment in the doc test itself, but it is redundant.

File: src/keys/secret.rs [hunk 4]
```diff
@@ -4,17 +4,48 @@
...
+/// use dusk_bls12_381::BlsScalar;
+/// use rand::rngs::StdRng;
+/// use rand::SeedableRng;
+///
+/// let mut rng = StdRng::seed_from_u64(12345);
+/// let note_secret_key = NoteSecretKey::random(&mut rng);
+/// let message = BlsScalar::uni_random(&mut rng);
+///
+/// let signature = note_secret_key.sign_single(&mut rng, message);
+/// let double_signature = note_secret_key.sign_double(&mut rng, message);
```
[the remark this answers is not in the document]
Comment (HDauven) [reply src/keys/secret.rs:46]: This was to showcase the functions available in this module as well, but like you say you quickly end up with a ton of duplication. Will turn it into the minimum required for the struct creation.

**Review by Neotamandua — APPROVED** [review]
well done, LGTM

**Pull request merged**